oci/private/tarball.bzl

```
@@ -98,20 +127,30 @@ def _tarball_impl(ctx):
         substitutions = substitutions,
     )
 
-    # TODO(2.0): this oci_tarball rule should just produce an mtree manifest instead,
-    # and then the tar rule can be composed in the oci_tarball macro in defs.bzl.
-    # To make it a non-breaking change, call the tar program from within this action instead.
+    # inputs both for creating the mtree spec and also invoking `tar --create`
+    tar_inputs = depset(
+        direct = [image, repo_tags, executable],
+        transitive = [bsdtar.default.files],
+    )
+    mtree_outputs = [mtree_spec, image_json]
     ctx.actions.run(
         executable = util.maybe_wrap_launcher_for_windows(ctx, executable),
-        inputs = depset(
-            direct = [image, repo_tags, executable],
-            transitive = [bsdtar.default.files],
-        ),
-        outputs = [tarball],
+        inputs = tar_inputs,
+        outputs = mtree_outputs,
         tools = [jq.bin],
-        mnemonic = "OCITarball",
-        progress_message = "OCI Tarball %{label}",
-        toolchain = None,
+        mnemonic = "OCITarballManifest",
+    )
+
+    tar_args = ctx.actions.args()
+    tar_args.add_all(["--create", "--no-xattr", "--no-mac-metadata"])
+    tar_args.add("--file", tarball)
+    tar_args.add(mtree_spec, format = "@%s")
+    ctx.actions.run(
+        executable = bsdtar.tarinfo.binary,
+        inputs = depset(direct = mtree_outputs, transitive = [tar_inputs]),
+        outputs = [tarball],
+        arguments = [tar_args],
+        mnemonic = "Tar",
```

```suggestion
        mnemonic = "OCITarball",
```

Or at least I'd mention OCI in the mnemonic somewhere - we've found we often want to treat "docker packaging (probably huge)" separately from "just tarring some random files" with things like `--modify_execution_info`